src/com/palmergames/bukkit/towny/object/TownSpawnLevel.java

```diff
@@ -7,63 +7,117 @@
 			ConfigNodes.GTOWN_SETTINGS_ALLOW_TOWN_SPAWN_TRAVEL,
 			"msg_err_public_spawn_forbidden",
+			"msg_err_town_spawn_forbidden_war",
+			"msg_err_town_spawn_forbidden_peace",
 			ConfigNodes.ECO_PRICE_TOWN_SPAWN_TRAVEL_PUBLIC,
-			PermissionNodes.TOWNY_SPAWN_PUBLIC.getNode()), ADMIN(
+			PermissionNodes.TOWNY_SPAWN_PUBLIC.getNode()),
+	ADMIN(
+			null,
+			null,
 			null,
 			null,
 			null,
 			null);
 
 	private ConfigNodes isAllowingConfigNode, ecoPriceConfigNode;
-	private String permissionNode, notAllowedLangNode;
+	private String permissionNode, notAllowedLangNode, notAllowedLangNodeWar, notAllowedLangNodePeace;
 
-	private TownSpawnLevel(ConfigNodes isAllowingConfigNode, String notAllowedLangNode, ConfigNodes ecoPriceConfigNode, String permissionNode) {
+	private TownSpawnLevel(ConfigNodes isAllowingConfigNode, String notAllowedLangNode, String notAllowedLangNodeWar, String notAllowedLangNodePeace, ConfigNodes ecoPriceConfigNode, String permissionNode) {
 
 		this.isAllowingConfigNode = isAllowingConfigNode;
 		this.notAllowedLangNode = notAllowedLangNode;
+		this.notAllowedLangNodeWar = notAllowedLangNodeWar;
+		this.notAllowedLangNodePeace = notAllowedLangNodePeace;
 		this.ecoPriceConfigNode = ecoPriceConfigNode;
 		this.permissionNode = permissionNode;
 	}
 
-	public void checkIfAllowed(Towny plugin, Player player) throws TownyException {
+	public void checkIfAllowed(Towny plugin, Player player, Town town) throws TownyException {
 
-		if (!(isAllowed() && hasPermissionNode(plugin, player)))
+		if (!(isAllowed(town) && hasPermissionNode(plugin, player, town))) {
+			boolean war = TownyUniverse.isWarTime() || TownyWar.isUnderAttack(town);
+			SpawnLevel level = SpawnLevel.parseSpawnLevel(this.isAllowingConfigNode.getRoot());
+			if(level == SpawnLevel.WAR && !war) {
+				throw new TownyException(TownySettings.getLangString(notAllowedLangNodeWar));
+			}
+			else if(level == SpawnLevel.PEACE && war) {
+				throw new TownyException(TownySettings.getLangString(notAllowedLangNodePeace));
+			}
 			throw new TownyException(TownySettings.getLangString(notAllowedLangNode));
+		}
 	}
 
-	public boolean isAllowed() {
+	public boolean isAllowed(Town town) {
 
-		return this == TownSpawnLevel.ADMIN ? true : TownySettings.getBoolean(this.isAllowingConfigNode);
+		return this == TownSpawnLevel.ADMIN ? true : isAlllowedTown(town);
 	}
 
-	public boolean hasPermissionNode(Towny plugin, Player player) {
+	public boolean hasPermissionNode(Towny plugin, Player player, Town town) {
 
-		return this == TownSpawnLevel.ADMIN ? true : (plugin.isPermissions() && TownyUniverse.getPermissionSource().has(player, this.permissionNode)) || ((!plugin.isPermissions()) && (TownySettings.isAllowingTownSpawn()));
+		return this == TownSpawnLevel.ADMIN ? true : (plugin.isPermissions() && TownyUniverse.getPermissionSource().has(player, this.permissionNode)) || ((!plugin.isPermissions()) && (isAlllowedTown(town)));
+	}
+	
+	private boolean isAlllowedTown(Town town)

```

Comment:
one too many l's in isAllowedTown

Reply:
i know what you mean, but before it was also checked with "((!plugin.isPermissions()) && (TownySettings.isAllowingTownSpawn())"